cmake/modules/SearchRootCoreDeps.cmake

```diff
@@ -14,77 +14,100 @@ if(pyroot_experimental)
 endif()
 
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
-  # - Check if the PYTHON_EXECUTABLE deprecated variable was passed by the
-  # user; if so, check weather it points to Python 2 or 3 and set the
-  # appropriate Python{X}_EXECUTABLE variable
-  # - Look for Python3 and set the deprecated variables to the ones set
-  # automatically by find_package(Python3 ...)
-  # - Look for Python2 and set the deprecated variables to the ones set
-  # automatically by find_package(Python2 ...) ONLY IF PYTHON3 WASN'T FOUND
-  if(PYTHON_EXECUTABLE)
-    execute_process(COMMAND ${PYTHON_EXECUTABLE} -c "import sys;print(sys.version_info[0])"
+
+  if(PYTHON_EXECUTABLE OR Python_EXECUTABLE)
+    # - Check if the PYTHON_EXECUTABLE deprecated variable or Python_EXECUTABLE
+    # was passed by the user; if so, PyROOT will be built with a single version
+    if(PYTHON_EXECUTABLE)
+      set(Python_EXECUTABLE "${PYTHON_EXECUTABLE}")
+    endif()
+    execute_process(COMMAND ${Python_EXECUTABLE} -c "import sys;print(sys.version_info[0])"
                     OUTPUT_VARIABLE PYTHON_PREFER_VERSION
                     ERROR_VARIABLE PYTHON_PREFER_VERSION_ERR)
     if(PYTHON_PREFER_VERSION_ERR)
-      message(WARNING "Unable to determine version of ${PYTHON_EXECUTABLE}: ${PYTHON_PREFER_VERSION_ERR}")
+      message(WARNING "Unable to determine version of ${Python_EXECUTABLE}: ${PYTHON_PREFER_VERSION_ERR}")
     endif()
     string(STRIP "${PYTHON_PREFER_VERSION}" PYTHON_PREFER_VERSION)
-    set(Python${PYTHON_PREFER_VERSION}_EXECUTABLE "${PYTHON_EXECUTABLE}")
-  endif()
+    find_package(Python COMPONENTS Interpreter Development NumPy)
+    if(Python_Development_FOUND)
+      set(PYTHON_EXECUTABLE "${Python_EXECUTABLE}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_INCLUDE_DIRS "${Python_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_LIBRARIES "${Python_LIBRARIES}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_STRING "${Python_VERSION}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MAJOR "${Python_VERSION_MAJOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_VERSION_MINOR "${Python_VERSION_MINOR}" CACHE INTERNAL "" FORCE)
+      set(PYTHON_UNDER_VERSION_STRING "${Python_VERSION_MAJOR}_${Python_VERSION_MINOR}")
+      set(NUMPY_FOUND ${Python_NumPy_FOUND} CACHE INTERNAL "" FORCE)
+      set(NUMPY_INCLUDE_DIRS "${Python_NumPy_INCLUDE_DIRS}" CACHE INTERNAL "" FORCE)
+    endif()
+    if(DEFINED Python_VERSION AND "${Python_VERSION}" VERSION_LESS "2.7")
+      message(WARNING "Ignoring Python installation: unsupported version ${Python_VERSION} (version>=2.7 required)")
+    endif()
+    if(Python_Development_FOUND)
+      message(STATUS "Main Python used to build: ${PYTHON_VERSION_MAJOR}; PyROOT built for version ${PYTHON_VERSION_MAJOR} ")
+    else()
+      message(WARNING "No supported Python development package was found; PyROOT will not be built.")
```

Comment:
This should be something like `No supported Python development package was found for the specified Python executable; PyROOT will not be built.`